test/unittest/rtps/builtin/BuiltinDataSerializationTests.cpp

```diff
@@ -2186,6 +2186,181 @@ TEST(BuiltinDataSerializationTests, interoperability_with_intercomdds)
     }
 }
 
+/*!
+ * This test checks that Fast DDS can properly serialize ResourceLimitsQos in DATA(w) when it has more fields than
+ * max_samples, max_instances and max_samples_per_instance. It also checks that those fields are not serialized
+ * into the allocated_samples and extra samples fields.
+ */
+TEST(BuiltinDataSerializationTests, interoperability_with_intercomdds_writer_resource_limits)
+{
+    const VendorId_t intercom_vendor_id = { 1, 5 };
+    // DATA(w)
+    {
+        // This was captured with wireshark from intercom_dds-3.16.2.0_shape_main_linux taken from
+        // https://github.com/omg-dds/dds-rtps/releases/tag/v1.2.2024
+        octet data_w_buffer[] =
+        {
+            // Encapsulation
+            0x00, 0x03, 0x00, 0x00,
+            // Writer GUID
```

Comment:
I would put the minimal required set of PIDs here. If we can make the deserialization pass with just the resource limits, I would do that.